src/sage/schemes/elliptic_curves/ell_number_field.py

```diff
@@ -4067,3 +4073,183 @@ def rational_points(self, **kwds):
+            not using the methods implemented here.
+
+            There are currently no theoretical results which allow
+            modularity to be proved over fields other than totally
+            real fields and imaginary quadratic fields.
+
+        ALGORITHM:
+
+        This is based on code provided by S. Siksek for the totally
+        real case, and relies on theorems in the following papers:
+        [Chen1996]_, [FLHS2015]_, [Thorne2016]_, [CarNew2023]_,
+        [DNS2020]_, [Box2022]_.
+
+        It relies on checking that the images of the mod-`p` Galois
+        representation attached to the elliptic curve for `p=3,5,7`
+        are not simultaneously small, except for the cases (base field
+        `\QQ`, real quadratic, totally real cubic, or totally real quartic
+        not containing `\sqrt{5}`) where theoretical results show this to
+        be impossible.
+
+        EXAMPLES:
+
+        Set ``verbose=True`` to see a reason for the conclusion::
+
+            sage: E = EllipticCurve('11a1')
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over QQ are modular
+            True
+            sage: K.<a> = QuadraticField(5)
+            sage: E = EllipticCurve([0,1,0,a,0])
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over real quadratic fields are modular
+            True
+            sage: E = EllipticCurve([0,0,0,a,0])
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over real quadratic fields are modular
+            True
+            sage: K.<a> = CyclotomicField(5)
+            sage: E = EllipticCurve([0,1,0,a,0])
+            sage: E.is_modular(verbose=True)
+            Unable to determine modularity except over totally real and imaginary quadratic fields
+            False
+
+        Some examples from the LMFDB.  Over a totally real quintic field::
+
+            sage: R.<x> = PolynomialRing(QQ)
+            sage: K.<a> = NumberField(R([1, 3, -1, -5, 0, 1]))
+            sage: E = EllipticCurve([K([3,0,-5,0,1]),K([-3,-7,9,2,-2]),K([1,0,0,0,0]),K([-22,-79,-92,14,21]),K([12,-87,-324,7,74])])
+            sage: E.is_modular(verbose=True)
+            Modular since the mod 3 Galois image is neither reducible nor split
+            True
+
+        Over imaginary quadratic fields `K` over which `X_0(15)` has
+        positive rank, there are many curves where current results do
+        not imply modularity::
+
+            sage: R.<x> = PolynomialRing(QQ)
+            sage: K.<a> = NumberField(R([2, -1, 1]))
+            sage: EllipticCurve('15a1').change_ring(K).rank()
+            1
+            sage: E = EllipticCurve([K([1,0]),K([1,-1]),K([1,1]),K([-105,98]),K([-615,1188])])
+            sage: E.is_modular(verbose=True)
+            Modularity not established: this curve has small image at 3 and 5
+            False
+            sage: EllipticCurve('15a1').change_ring(K).rank()
```

Comment:
```suggestion
            sage: EllipticCurve('15a1').change_ring(K).rank()  # long time
```

On a fast machine (i7-9700 CPU @ 3.00GHz):
```
Warning, slow doctest:
    EllipticCurve('15a1').change_ring(K).rank()
Test ran for 0.72 s, check ran for 0.00 s
```

This is borderline (0.72s is less than 1s, although on a slower/server machine/CI it will be > 1s).  But since this is not really a doctest to the current method, I don't see any harm in marking it long time.

Reply:
OK, but I did not know that 1s was the criterion for long time!